tests/drivers/build_all/w1 fails on heltec_{wifi_lora32,wireless_stick_lite_v3}/esp32/appcpu

**Describe the bug**
tests/drivers/build_all/w1 fails in CI in heltec_wifi_lora32/esp32/appcpu  or heltec_wireless_stick_lite_v3/esp32/appcpu  
The test is currently runnable in these platforms and triggers in CI

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2.
    * cmake -GNinja -DBOARD=heltec_wifi_lora32_v2/esp32/appcpu ../tests/drivers/build_all/w1/
OR 
    * cmake -GNinja -DBOARD=heltec_wireless_stick_lite_v3/esp32s3/appcpu  ../tests/drivers/build_all/w1/
4. ninja
5. See error

**Expected behavior**
No build errors

**Impact**
* Blocking CI when this combination is triggered.

**Logs and console output**
https://github.com/zephyrproject-rtos/zephyr/actions/runs/8299083419/job/22715109890#step:12:493
(see below)

**Environment (please complete the following information):**
 - OS: Linux (Ubuntu 22.04 / CI)
 - Toolchain: Zephyr SDK
 - Commit SHA or Version used: Today's main 4f3db943b9b3ab8616654f1b8b48acf74f06fd0e

**Additional context**
These targets were introduced in 8dc3f856229ce083c956aa301c31a23e65bd8cd8

-----

```
In file included from zephyr/include/zephyr/device.h:12,
                 from zephyr/boards/heltec/heltec_wifi_lora32_v2/board_init.c:7:
zephyr/boards/heltec/heltec_wifi_lora32_v2/board_init.c: In function 'board_heltec_wifi_lora32_v2_init':
zephyr/include/zephyr/devicetree.h:198:36: error: 'DT_N_NODELABEL_vext_P_gpios_IDX_0_VAL_pin' undeclared (first use in this function)
  198 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
zephyr/include/zephyr/devicetree.h:4462:9: note: in definition of macro 'DT_CAT7'
 4462 |         a1 ## a2 ## a3 ## a4 ## a5 ## a6 ## a7
      |         ^~
zephyr/include/zephyr/devicetree/gpio.h:110:9: note: in expansion of macro 'DT_PHA_BY_IDX'
  110 |         DT_PHA_BY_IDX(node_id, gpio_pha, idx, pin)
      |         ^~~~~~~~~~~~~
zephyr/include/zephyr/devicetree/gpio.h:121:9: note: in expansion of macro 'DT_GPIO_PIN_BY_IDX'
  121 |         DT_GPIO_PIN_BY_IDX(node_id, gpio_pha, 0)
      |         ^~~~~~~~~~~~~~~~~~
zephyr/boards/heltec/heltec_wifi_lora32_v2/board_init.c:11:19: note: in expansion of macro 'DT_GPIO_PIN'
   11 | #define VEXT_PIN  DT_GPIO_PIN(DT_NODELABEL(vext), gpios)
      |                   ^~~~~~~~~~~
zephyr/include/zephyr/devicetree.h:198:29: note: in expansion of macro 'DT_CAT'
  198 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
zephyr/boards/heltec/heltec_wifi_lora32_v2/board_init.c:11:31: note: in expansion of macro 'DT_NODELABEL'
   11 | #define VEXT_PIN  DT_GPIO_PIN(DT_NODELABEL(vext), gpios)
      |                               ^~~~~~~~~~~~
zephyr/boards/heltec/heltec_wifi_lora32_v2/board_init.c:24:34: note: in expansion of macro 'VEXT_PIN'
   24 |         gpio_pin_configure(gpio, VEXT_PIN, GPIO_OUTPUT);
      |                                  ^~~~~~~~
zephyr/include/zephyr/devicetree.h:198:36: note: each undeclared identifier is reported only once for each function it appears in
  198 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
zephyr/include/zephyr/devicetree.h:4462:9: note: in definition of macro 'DT_CAT7'
 4462 |         a1 ## a2 ## a3 ## a4 ## a5 ## a6 ## a7
      |         ^~
zephyr/include/zephyr/devicetree/gpio.h:110:9: note: in expansion of macro 'DT_PHA_BY_IDX'
  110 |         DT_PHA_BY_IDX(node_id, gpio_pha, idx, pin)
      |         ^~~~~~~~~~~~~
zephyr/include/zephyr/devicetree/gpio.h:121:9: note: in expansion of macro 'DT_GPIO_PIN_BY_IDX'
  121 |         DT_GPIO_PIN_BY_IDX(node_id, gpio_pha, 0)
      |         ^~~~~~~~~~~~~~~~~~
zephyr/boards/heltec/heltec_wifi_lora32_v2/board_init.c:11:19: note: in expansion of macro 'DT_GPIO_PIN'
   11 | #define VEXT_PIN  DT_GPIO_PIN(DT_NODELABEL(vext), gpios)
      |                   ^~~~~~~~~~~
zephyr/include/zephyr/devicetree.h:198:29: note: in expansion of macro 'DT_CAT'
  198 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
zephyr/boards/heltec/heltec_wifi_lora32_v2/board_init.c:11:31: note: in expansion of macro 'DT_NODELABEL'
   11 | #define VEXT_PIN  DT_GPIO_PIN(DT_NODELABEL(vext), gpios)
      |                               ^~~~~~~~~~~~
zephyr/boards/heltec/heltec_wifi_lora32_v2/board_init.c:24:34: note: in expansion of macro 'VEXT_PIN'
   24 |         gpio_pin_configure(gpio, VEXT_PIN, GPIO_OUTPUT);
      |                                  ^~~~~~~~
zephyr/include/zephyr/devicetree.h:198:36: error: 'DT_N_NODELABEL_oledrst_P_gpios_IDX_0_VAL_pin' undeclared (first use in this function)
  198 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
zephyr/include/zephyr/devicetree.h:4462:9: note: in definition of macro 'DT_CAT7'
 4462 |         a1 ## a2 ## a3 ## a4 ## a5 ## a6 ## a7
      |         ^~
zephyr/include/zephyr/devicetree/gpio.h:110:9: note: in expansion of macro 'DT_PHA_BY_IDX'
  110 |         DT_PHA_BY_IDX(node_id, gpio_pha, idx, pin)
      |         ^~~~~~~~~~~~~
zephyr/include/zephyr/devicetree/gpio.h:121:9: note: in expansion of macro 'DT_GPIO_PIN_BY_IDX'
  121 |         DT_GPIO_PIN_BY_IDX(node_id, gpio_pha, 0)
      |         ^~~~~~~~~~~~~~~~~~
zephyr/boards/heltec/heltec_wifi_lora32_v2/board_init.c:12:19: note: in expansion of macro 'DT_GPIO_PIN'
   12 | #define OLED_RST  DT_GPIO_PIN(DT_NODELABEL(oledrst), gpios)
      |                   ^~~~~~~~~~~
zephyr/include/zephyr/devicetree.h:198:29: note: in expansion of macro 'DT_CAT'
  198 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
zephyr/boards/heltec/heltec_wifi_lora32_v2/board_init.c:12:31: note: in expansion of macro 'DT_NODELABEL'
   12 | #define OLED_RST  DT_GPIO_PIN(DT_NODELABEL(oledrst), gpios)
      |                               ^~~~~~~~~~~~
zephyr/boards/heltec/heltec_wifi_lora32_v2/board_init.c:28:34: note: in expansion of macro 'OLED_RST'
   28 |         gpio_pin_configure(gpio, OLED_RST, GPIO_OUTPUT);
      |                                  ^~~~~~~~
```

--- a/tests/drivers/build_all/w1/testcase.yaml
+++ b/tests/drivers/build_all/w1/testcase.yaml
@@ -7,5 +7,7 @@ tests:
     tags:
       - drivers
       - w1
+    depends_on:
+      - gpio
     integration_platforms:
       - native_posix

